.github/workflows/docker-build.yaml

```diff
@@ -29,18 +32,28 @@ jobs:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
 
-      - name: Build and push
+      - name: Build
         uses: docker/build-push-action@v3
         with:
           context: .
-          push: ${{ github.ref == 'refs/heads/master' }}
+          outputs: type=docker,dest=/tmp/docker.tar
           tags: ${{ secrets.DOCKERHUB_USERNAME }}/bancho.py:latest
           cache-from: type=gha
           cache-to: type=gha,mode=max
-          outputs: type=docker,dest=/tmp/docker.tar
 
       - name: Upload artifact
         uses: actions/upload-artifact@v3
         with:
           name: docker
           path: /tmp/docker.tar
+
+      - name: Push to Docker Hub
+        uses: docker/build-push-action@v3
+        if: ${{ github.ref == 'refs/heads/master' }}
+        with:
+          context: .
+          platforms: $PLATFORMS
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/bancho.py:latest
+          cache-from: type=gha
+          cache-to: type=gha,mode=max

```

Comment:
will this trigger a 2nd build? does it use the artifact?

Reply:
have followed https://docs.docker.com/build/ci/github-actions/test-before-push/

it will use the cached version

Reply:
i see that's using `docker/build-push-action@v5` while we're using v3

should we upgrade?

Reply:
oh, did not notice that. yeah we should upgrade with all the recent github changes with actions and node version

Reply:
updated versions